components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -264,6 +264,12 @@ abstract class AbstractFetchDownloadService : Service() {
             useFileStream(newDownloadState, isResumingDownload) { outStream ->
                 copyInChunks(downloadJobs[download.id]!!, inStream, outStream)
             }
+
+            downloadJobs[download.id]?.let {
```

Comment:
`copyInChunks` loops as long as `status` is `ACTIVE`. So, we break out of the loop when `pausing` the download. However, with this change, we may set the `status` to failed for paused downloads if the job isn't cancelled fast enough. I think we need to add an additional check here to see if the download was paused. We also need to make sure changes / writes to `status` are visible across threads.